deployment/ccip/changeset/cs_rmn_curse_uncurse.go

```diff
@@ -0,0 +1,275 @@
+
+	for _, curseAction := range c.CurseActions {
+		result := curseAction(e)
+		for _, action := range result {
+			targetChain := e.Chains[action.ChainSelector]
+			targetChainState := state.Chains[action.ChainSelector]
+
+			if err := commoncs.ValidateOwnership(e.GetContext(), c.MCMS != nil, targetChain.DeployerKey.From, targetChainState.Timelock.Address(), targetChainState.RMNRemote); err != nil {
+				return fmt.Errorf("chain %s: %w", targetChain.String(), err)
+			}
+
+			if !slices.Contains(validSelectors, action.ChainSelector) {
+				return errors.Errorf("invalid chain selector %d for chain %s", action.ChainSelector, targetChain.String())
+			}
+
+			if !slices.Contains(validSubjects, action.SubjectToCurse) {
+				return errors.Errorf("invalid subject %x for chain %s", action.SubjectToCurse, targetChain.String())
+			}
+		}
+	}
+
+	return nil
+}
+
+type Subject = [16]byte
+
+func SelectorToSubject(subject uint64) Subject {
+	var b Subject
+	binary.BigEndian.PutUint64(b[8:], subject)
+	return b
+}
+
+// CurseLaneOnlyOnSource curses a lane only on the source chain
+// This will prevent message from source to destination to be initiated
+// One noteworthy behaviour is that this means that message can be sent from destination to source but will not be executed on the source
+func CurseLaneOnlyOnSource(sourceSelector uint64, destinationSelector uint64) CurseAction {
+	// Curse from source to destination
+	return func(e deployment.Environment) []RMNCurseAction {
+		return []RMNCurseAction{
+			{
+				ChainSelector:  sourceSelector,
+				SubjectToCurse: SelectorToSubject(destinationSelector),
+			},
+		}
+	}
+}
+
+func CurseLane(sourceSelector uint64, destinationSelector uint64) CurseAction {
+	// Bidirectional curse between two chains
+	return func(e deployment.Environment) []RMNCurseAction {
+		return append(
+			CurseLaneOnlyOnSource(sourceSelector, destinationSelector)(e),
+			CurseLaneOnlyOnSource(destinationSelector, sourceSelector)(e)...,
+		)
+	}
+}
+
+func CurseChain(chainSelector uint64) CurseAction {
+	return func(e deployment.Environment) []RMNCurseAction {
+		chainSelectors := e.AllChainSelectors()
+
+		// Curse all other chains to prevent onramp from sending message to the cursed chain
+		var curseActions []RMNCurseAction
+		for _, otherChainSelector := range chainSelectors {
+			if otherChainSelector != chainSelector {
+				curseActions = append(curseActions, RMNCurseAction{
+					ChainSelector:  otherChainSelector,
+					SubjectToCurse: SelectorToSubject(chainSelector),
+				})
+			}
+		}
+
+		// Curse the chain with a global curse to prevent any onramp or offramp message from send message in and out of the chain
+		curseActions = append(curseActions, RMNCurseAction{
+			ChainSelector:  chainSelector,
+			SubjectToCurse: GlobalCurseSubject(),
+		})
+
+		return curseActions
+	}
+}
+
+func groupRMNSubjectBySelector(rmnSubjects []RMNCurseAction, filter bool) map[uint64][]Subject {
```

Comment:
```suggestion
func groupRMNSubjectBySelector(rmnSubjects []RMNCurseAction, keepOnlyGlobal bool) map[uint64][]Subject {
```

Based on my understanding of this code, this seems like a more meaningful var name, please correct me if I'm wrong though

Reply:
I changed the function to have less confusing options but keepOnlyGlobal would not be enough as we also need to remove the self curses I was mentioning above.